Add make verify target for running static analysis checks

henrybear327 opened the pull request:
Add Makefile targets for running fmt and static analysis checks

Changes:
- Introduce `make verify` and `make fmt` targets in Makefile
- For linter, use the same config as etcd repo
- Update GitHubCI workflow accordingly 

Closes #31

References:
- [fmt and lint](https://github.com/etcd-io/bbolt/blob/c57b3537ed933f2e613f04d42688d49e511f69b3/Makefile#L30C1-L36C84) 
- [fmt script](https://github.com/etcd-io/etcd/blob/a08812eb01572aa64479543d1e8fa33cda22b8b8/scripts/fix.sh)
- [GithubCI](https://github.com/etcd-io/etcd/blob/a08812eb01572aa64479543d1e8fa33cda22b8b8/.github/workflows/static-analysis.yaml)
- [linter config](https://github.com/etcd-io/etcd/blob/a08812eb01572aa64479543d1e8fa33cda22b8b8/tools/.golangci.yaml)

henrybear327 commented:
@jmhbnz do we want to update the `static-analysis` pipeline to use `make verify`? 

Also, would we want to include the [`fmt` part](https://github.com/etcd-io/bbolt/blob/main/Makefile#L30) for house-keeping, too?

siyuanfoundation commented:
> @jmhbnz do we want to update the `static-analysis` pipeline to use `make verify`?
> 
> Also, would we want to include the [`fmt` part](https://github.com/etcd-io/bbolt/blob/main/Makefile#L30) for house-keeping, too? :)

I think yes for both :) 

Thanks!

jmhbnz commented:
> > @jmhbnz do we want to update the `static-analysis` pipeline to use `make verify`?
> > Also, would we want to include the [`fmt` part](https://github.com/etcd-io/bbolt/blob/main/Makefile#L30) for house-keeping, too? :)
> 
> I think yes for both :)

Agreed - Thanks @henrybear327 

Note: Please ensure your commit is signed so the developer certificate of origin check passes.

henrybear327 commented:
> > @jmhbnz do we want to update the `static-analysis` pipeline to use `make verify`?
> > Also, would we want to include the [`fmt` part](https://github.com/etcd-io/bbolt/blob/main/Makefile#L30) for house-keeping, too? :)
> 
> I think yes for both :)
> 
> Thanks!

Thanks for the feedback! :) 

I have updated the linter target and added `fmt`! 

For the linter, I took the [linter config file](https://github.com/etcd-io/etcd/blob/a08812eb01572aa64479543d1e8fa33cda22b8b8/tools/.golangci.yaml) from `etcd` project, as I am trying to keep the consistency across projects :)

siyuanfoundation commented on scripts/fix.sh:
```diff
@@ -0,0 +1,13 @@
+GO_CMD="go"
+
+# TODO(ptabor): Expand to cover different architectures (GOOS GOARCH), or just list go files.
```
Is this TODO needed?

henrybear327 replied:
I didn't modify the file when I took it from the etcd repo. :( But thanks for pointing this out, I will remove it to keep it clean! :) 

Thank you! :)

siyuanfoundation commented:
Don't forget to signoff every commit :)

siyuanfoundation commented:
One nit comment. Otherwise LGTM.
